tests/backup/backup_psa_test.go

```diff
@@ -28,3 +41,605 @@ var _ = Describe("{DummyPSATestcase}", Label(TestCaseLabelsMap[DummyPSATestcase]
+			log.InfoD("Taking schedule backup of namespaces with rules")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			log.InfoD("Taking schedule backup of applications with both volume and resources on namespace with baseline label set")
+			// backupShedNameWithVolumeAndResourceWithRestrictedLabel := fmt.Sprintf("%s-%v", BackupNamePrefix, RandomString(10))
+
+			scheduleName := fmt.Sprintf("%s-schedule-with-rules-%s", BackupNamePrefix, RandomString(4))
+			scheduleBackupName, err := CreateScheduleBackupWithValidation(ctx, scheduleName, SourceClusterName, backupLocationName, backupLocationUID, baselineScheduledAppContexts, nil, BackupOrgID, preRuleName, preRuleUid, postRuleName, postRuleUid, periodicSchedulePolicyName, periodicSchedulePolicyUid)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of backup [%s]", scheduleBackupName))
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup of namespaces with volume and resources [%s]", backupNameWithVolumeAndResource))
+			err = SuspendBackupSchedule(scheduleName, periodicSchedulePolicyName, BackupOrgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Suspending Backup Schedule [%s] ", scheduleName))
+			scheduleNames = append(scheduleNames, scheduleName)
+			backupNamespaceMap[scheduleBackupName] = baselineNamespaceInSrcCluster
+			backupList = append(backupList, scheduleBackupName)
+
+			//  Taking restore of the backup taken
+			log.InfoD("Restoring backup of restricted namespace with namespacemapping")
+			defaultRestoreName = fmt.Sprintf("%s-%v-ns-mapping-%v", RestoreNamePrefix, RandomString(3), scheduleBackupName)
+			//appContextsToRestore := FilterAppContextsByNamespace(scheduledAppContexts, appNamespaces)
+			err = CreateRestoreWithValidation(ctx, defaultRestoreName, scheduleBackupName, nil, nil, DestinationClusterName, BackupOrgID, baselineScheduledAppContexts)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying %s backup's restore %s creation with namespace mapping %v and storage class mapping %v", scheduleBackupName, defaultRestoreName, nil, nil))
+			restoreList = append(restoreList, defaultRestoreName)
+		})
+
+		JustAfterEach(func() {
+			defer EndPxBackupTorpedoTest(scheduledAppContexts)
+			defer func() {
+				Inst().AppList = defaultAppList
+			}()
+			opts := make(map[string]bool)
+			opts[SkipClusterScopedObjects] = true
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, backup := range backupList {
+				backupUid, err := Inst().Backup.GetBackupUID(ctx, backup, BackupOrgID)
+				log.FailOnError(err, "Unable to fetch backup UID")
+				_, err = DeleteBackup(backup, backupUid, BackupOrgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting backup [%s]", backup))
+			}
+			log.Info("Destroying scheduled apps on source cluster")
+			DestroyApps(scheduledAppContexts, opts)
```

Comment:
Can we put this in a loop like:
```
for _, context := range contexts{
         DestroyApps(context, opts)
}
```

Reply:
But, I believe DestroyApps func takes array of contexts

Reply:
we can have a nested array of contexts right?